pages/directory/index.tsx

```diff
@@ -1,28 +1,106 @@
-import { Box, Container, List, ListItem, Typography } from '@mui/material';
+import { Box, Container, CircularProgress, List, ListItem, Typography } from '@mui/material';
+import axios from 'axios';
 import { NextPage } from 'next';
-import React from 'react';
+import * as React from 'react';
 
+import ErrorPopup from 'client/components/ErrorPopup';
 import createAxiosInstance from 'client/lib/axios';
 import { ClientUser } from 'shared/User';
 
-const Directory: NextPage<{ users: ClientUser[]; error: string }> = props => {
-  const { users, error } = props;
+const Directory: NextPage<{ users: ClientUser[]; totalPages: number; error: string }> = props => {
+  const { users, totalPages, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPageNum, setTotalPageNum] = React.useState(totalPages);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [showError, setShowError] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  async function fetchUsers(page: number) {
+    try {
+      const response = await axios.get(`/api/users?page=${page}`);
+      return {
+        page: response.data.page,
+        total: response.data.totalPages,
+        users: response.data.users,
+      };
+    } catch (err) {
+      if (err.code === 'ERR_BAD_REQUEST') {
```

Comment:
What happens if other errors occur? Also, is `ERR_BAD_REQUEST` only going to occur when the requested page is out of range?

Reply:
I didn't know all the errors that were possible, Ill look into it more

Reply:
Every possible error doesn't need to be explicitly handled; the user doesn't always need to know what the error was.

Some errors may be network related and work themselves out, others might be things we need to address... either way, communicating that an error has occurred to the user instead of leaving them uncertain results in a better user experience.